server/src/test/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryTests.java

```diff
@@ -813,14 +861,15 @@ public void testDeleteStaleCommitsActualDeleteIOException() throws IOException {
         doThrow(new IOException("Error")).when(remoteDataDirectory).deleteFile(segmentFileWithException);
         // popluateMetadata() adds stub to return 3 metadata files
         // We are passing lastNMetadataFilesToKeep=2 here so that oldest 1 metadata file will be deleted
-        remoteSegmentStoreDirectory.deleteStaleSegments(2);
+        remoteSegmentStoreDirectory.deleteStaleSegmentsAsync(2);
```

Comment:
If the asserts below depends on delete to complete, there is a possibility of flaky test.